include/irr/core/math/morton.h

```diff
@@ -29,29 +29,54 @@ namespace impl
     template <typename T>
     inline T morton2d_decode(T x)
     {
-        x = x & morton2d_decode_mask<T>(0);
-        x = (x | (x >> 1)) & morton2d_decode_mask<T>(1);
-        x = (x | (x >> 2)) & morton2d_decode_mask<T>(2);
+        x = x & morton2d_mask<T>(0);
+        x = (x | (x >> 1)) & morton2d_mask<T>(1);
+        x = (x | (x >> 2)) & morton2d_mask<T>(2);
         IRR_PSEUDO_IF_CONSTEXPR_BEGIN (sizeof(T)>1u)
         {
-        x = (x | (x >> 4)) & morton2d_decode_mask<T>(3);
+        x = (x | (x >> 4)) & morton2d_mask<T>(3);
         } IRR_PSEUDO_IF_CONSTEXPR_END
         IRR_PSEUDO_IF_CONSTEXPR_BEGIN (sizeof(T)>2u)
         {
-        x = (x | (x >> 8)) & morton2d_decode_mask<T>(4);
+        x = (x | (x >> 8)) & morton2d_mask<T>(4);
         } IRR_PSEUDO_IF_CONSTEXPR_END
         IRR_PSEUDO_IF_CONSTEXPR_BEGIN (sizeof(T)>4u)
         {
         x = (x | (x >> 16));
         } IRR_PSEUDO_IF_CONSTEXPR_END
         return x;
     }
+
+    //! Puts bits on even positions filling gaps with 0s
+    template <typename T>
+    inline T separate_bits(T x)
```

Comment:
this function is different for 2d and 3d I'd suppose

Reply:
ofc, i have only 2d functions

Reply:
yeah just make it explicit in the name that this is a `2d` function